src/cuda/rtc/compilation_options.hpp

```diff
@@ -485,8 +485,8 @@ struct opt_start_t {
 
 } // namespace detail_
 
-template <typename MarshalTarget, typename Delimiter>
-MarshalTarget& operator<<(MarshalTarget& mt, detail_::opt_start_t<Delimiter>& opt_start)
+template <typename Delimiter>
+std::ostream& operator<<(std::ostream& mt, detail_::opt_start_t<Delimiter>& opt_start)
```

Comment:
I'm not sure the MarshalTarget is always an `std::ostream`.

Reply:
Yup, in fact it is sometimes a `marshalled_options_t` object - that's the original marshalling we do, for passing the arguments to the C API call.